### PR TITLE
Add commands to allow setting of timeprop parameters

### PR DESCRIPTION
**Pull request by blacknell**
## Description:

Currently you have to define timeprop parameters using compile time directives. This makes managing multiple instances of tasmota more burdensome.

add: new commands to set timeprop settings (documented in the source code)
add: optionally include current settings in json output (default is `false` since these will rarely change)
fix: fallback power values are now properly handled as floats between 0.0 and 1.0
fix: typo in compilation directive for `USE_TIMEPROP`
chore: refactored the existing cmnd code to use the more convention command decoder
chore: removed redundant commented code for clarity

**Related issue (if applicable):** fixes # N/A

## Checklist:
  - [x] The pull request is done against the latest development branch
  - [x] Only relevant files were touched
  - [x] Only one feature/fix was added per PR and the code change compiles without warnings
  - [x] The code change is tested and works with Tasmota core ESP8266 V.2.7.4.9
  - [x] The code change is tested and works with Tasmota core ESP32 V.2.0.11
  - [x] I accept the [CLA](https://github.com/arendst/Tasmota/blob/development/CONTRIBUTING.md#contributor-license-agreement-cla).

_NOTE: The code change must pass CI tests. **Your PR cannot be merged unless tests pass**_

### CODE REVIEW

File: tasmota/tasmota_xdrv_driver/xdrv_48_timeprop.ino [hunk 1]
```diff
@@ -135,19 +166,103 @@ void TimepropSetPower(int index, float power) {
...
+// 'TimepropSetCycleTime0' will retrieve the value of cycle time for output 0
+
+void CmndSetCycleTime(void) {
+  if (XdrvMailbox.index >=0 && XdrvMailbox.index < TIMEPROP_NUM_OUTPUTS ) {
+    if(XdrvMailbox.data_len) {
+      int newCycleTime=TextToInt(XdrvMailbox.data);
+      if(newCycleTime>0) {
+        cycleTimes[XdrvMailbox.index] = newCycleTime;
+        Tprop.timeprops[XdrvMailbox.index].initialise(cycleTimes[XdrvMailbox.index], deadTimes[XdrvMailbox.index], opInverts[XdrvMailbox.index], fallbacks[XdrvMailbox.index], maxIntervals[XdrvMailbox.index], Tprop.current_time_secs);
+        Response_P(PSTR("{\"" D_CMND_TIMEPROP_PREFIX D_CMND_TIMEPROP_SETCYCLETIME "%d\":\"%d\"}"), XdrvMailbox.index, newCycleTime);
```
Comment (arendst) [inline tasmota/tasmota_xdrv_driver/xdrv_48_timeprop.ino:200]: Pls verify as JSON values should not be using quotes.

Comment (blacknell) [reply tasmota/tasmota_xdrv_driver/xdrv_48_timeprop.ino:200]: I did and it looks valid to me. I also put various output through a json validator.

Here's an example output
```
21:43:44.959 MQT: tele/tasmota_E7AD3C/SENSOR = {"Time":"2023-08-14T21:43:44","HDC1080":{"Temperature":22.6,"Humidity":100.0,"DewPoint":22.6},"ESP32":{"Temperature":26.1},"Timeprop":{"Output0":{"CycleTime":600,"DeadTime":0,"OutputInvert":0,"FallbackPower":0.00,"MaxUpdateInterval":120}},"PID":{"PidPv":22.70,"PidSp":9.00,"PidPb":5.00,"PidTi":1800.00,"PidTd":15.00,"PidInitialInt":0.50,"PidDSmooth":3.00,"PidAuto":1,"PidManualPower":0.00,"PidMaxInterval":300,"PidInterval":10,"PidUpdateSecs":0,"PidSensorLost":0,"PidPower":0.00},"TempUnit":"C"}
```

Comment (arendst) [reply tasmota/tasmota_xdrv_driver/xdrv_48_timeprop.ino:200]: You'll have to agree this is wrong:
```
Response_P(PSTR("{\"" D_CMND_TIMEPROP_PREFIX D_CMND_TIMEPROP_SETCYCLETIME "%d\":\"%d\"}"), XdrvMailbox.index, newCycleTime);
```
as it ends up like this:
```
{" D_CMND_TIMEPROP_PREFIX D_CMND_TIMEPROP_SETCYCLETIME 1":"123"}
```
The last quotes shouldn't be there as it's a number, not a string.

Comment (blacknell) [reply tasmota/tasmota_xdrv_driver/xdrv_48_timeprop.ino]: Ah, this is the json output as a result of the commands. I used the same code as was in the single command there previously for the new commands added. This code was always wrong, so I've now replaced with `ResponseCmndFloat` etc.

Sorry about that. I hadn't spotted and/or realised.

**Pull request merged**